client/my-sites/upgrades/checkout-thank-you/google-voucher-details.jsx

```diff
@@ -0,0 +1,240 @@
+/**
+ * External dependencies
+ */
+import React, { PropTypes, Component } from 'react';
+import { connect } from 'react-redux';
+import debounce from 'lodash/debounce';
+
+/**
+ * Internal dependencies
+ */
+import localize from 'lib/mixins/i18n/localize';
+import ClipboardButtonInput from 'components/clipboard-button-input';
+import PurchaseDetails from 'components/purchase-detail';
+import PurchaseButton from 'components/purchase-detail/purchase-button';
+import TipInfo from 'components/purchase-detail/tip-info';
+import Dialog from 'components/dialog';
+import googleTermsAndConditions from './google-voucher-TC';
+import QuerySiteVouchers from 'components/data/query-site-vouchers';
+import {
+	assignSiteVoucher as assignVoucher
+} from 'state/sites/vouchers/actions';
+import { GOOGLE_CREDITS } from 'state/sites/vouchers/service-types';
+import {
+	getVouchersBySite,
+	getGoogleAdCredits
+} from 'state/sites/vouchers/selectors';
+
+class GoogleVoucherDetails extends Component {
+	constructor() {
+		super();
+
+		// bounds
+		this.onButtonClick = this.onButtonClick.bind( this );
+		this.onDialogCancel = this.onDialogCancel.bind( this );
+		this.onAgreeButton = this.onAgreeButton.bind( this );
+
+		// debounced
+		this.incrementStep = debounce( this.incrementStep, 300 );
+
+		this.state = {
+			showTermsAndConditions: false,
+			step: 0
+		};
+	}
+
+	componentWillMount() {
+		const voucher = this.getVoucher();
+		if ( voucher && voucher.status === 'assigned' ) {
+			this.setState( { step: 2 } );
+		}
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		if ( ! nextProps.googleAdCredits ) {
+			return null;
+		}
+
+		if ( this.props.googleAdCredits === nextProps.googleAdCredits ) {
+			return null;
+		}
+
+		if ( nextProps.googleAdCredits.length > 0 ) {
+			this.setState( { step: 2 } );
+		}
+	}
+
+	onButtonClick() {
+		this.incrementStep();
+	}
+
+	onDialogCancel() {
+		this.setState( {
+			showTermsAndConditions: false,
+			step: 0
+		} );
+	}
+
+	onAgreeButton() {
+		this.props.assignVoucher( this.props.selectedSite.ID, GOOGLE_CREDITS );
+
+		this.setState( {
+			showTermsAndConditions: false,
+			step: 2
+		} );
+	}
+
+	incrementStep() {
+		this.setState( {
+			step: this.state.step + 1,
+			showTermsAndConditions: ( this.state.step + 1 ) === 1
+		} );
+	}
+
+	getVoucher( props = this.props ) {
+		const { googleAdCredits } = props;
+		return googleAdCredits.length > 0 ? googleAdCredits[0] : {};
+	}
+
+	renderInitialStep() {
+		return (
+			<div className="purchase-detail__body">
+				<PurchaseButton
+					onClick={ this.onButtonClick }
+					text={ this.props.translate( 'Generate Code' ) } />
+
+				<TipInfo
+					info={ this.props.translate( 'Offer valid in US and Canada after spending the first $25 on Google AdWords.' ) } />
+			</div>
+		);
+	}
+
+	renderTermsAndConditions() {
+		return (
+			<Dialog
+				isVisible={ this.state.showTermsAndConditions }
```

Comment:
As far as I can tell, `this.state.showTermsAndConditions` is always true while this block of code is being rendered. So the attribute seems unnecessary, and this could just be `isVisible={ true }`